pkg/prometheus/agent_http.go

```diff
@@ -0,0 +1,231 @@
+package prometheus
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/pkg/prometheus/configapi"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	totalCreatedConfigs = promauto.NewCounter(prometheus.CounterOpts{
```

Comment:
I think it would be useful to know the current number of configs on each agent. Which will help analyse load distribution, perhaps a gauge that we can increment /decrement on create/delete?

Reply:
Great idea. I don't think it's a good fir for this PR since a config added here won't necessarily go to the agent that received the request. 

I'll make sure I add this into the next PR that adds the clustering/sharding.

Reply:
Since I'm including the ConfigManager code in this PR now, I've now added the metric here.